### PR TITLE
Add paths-labeller workflow

### PR DESCRIPTION
**Pull request by brokensound77**
## Issues
part of elastic/security-team/issues/1151

## Summary
Create a botelastic paths labeller

### CODE REVIEW

File: .github/paths-labeller.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  - "rules/integrations/okta/**/*.toml"
+- "python":
+  - "detection_rules/**/*.py"
+- "RTA":
+  - "rta/**/*"
+- "Rule: Deprecation":
+  - "rules/_deprecated/**/*"
+# tests
+- "test-suite":
+  - "tests/**/*"
```
Comment (brokensound77) [inline .github/paths-labeller.yml:53]:
```suggestion
  - "tests/**/*"
  
```

File: .github/paths-labeller.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+- "OS: macOS":
+  - "rules/macos/**/*.toml"
+- "OS: Windows":
+  - "rules/windows/**/*.toml"
+- "Platform: AWS":
+  - "rules/integrations/aws/**/*.toml"
+- "Platform: Azure":
+  - "rules/integrations/azure/**/*.toml"
+- "Platform: Crowdstrike":
+    - "rules/integrations/crowdstrike/**/*.toml"
```
Comment (brokensound77) [inline .github/paths-labeller.yml:34]:
```suggestion
  - "rules/integrations/crowdstrike/**/*.toml"
```

File: .github/paths-labeller.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+- "Platform: CyberArkPas":
+    - "rules/integrations/cyberarkpas/**/*.toml"
```
Comment (brokensound77) [inline .github/paths-labeller.yml:36]:
```suggestion
  - "rules/integrations/cyberarkpas/**/*.toml"
```

File: .github/paths-labeller.yml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+- "python":
+  - "detection_rules/**/*.py"
+- "RTA":
+  - "rta/**/*"
+- "Rule: Deprecation":
+  - "rules/_deprecated/**/*"
+# tests
+- "test-suite":
+  - "tests/**/*"
+
```
Comment (brokensound77) [inline .github/paths-labeller.yml:54]:
```suggestion

```

File: .github/paths-labeller.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
+---
+- "beats":
+  - "detection_rules/beats.py"
+  - "etc/beats_schemas/**/*"
+- "documentation":
+  - "./**/*.md"
```
Comment (rw-access) [inline .github/paths-labeller.yml:6]: does this also match `.md` in the root folder?

File: .github/paths-labeller.yml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+- "Domain: Cloud":
+  - "rules/integrations/aws/**/*.toml"
+  - "rules/integrations/azure/**/*.toml"
+  - "rules/integrations/gcp/**/*.toml"
+  - "rules/integrations/google-workspace/**/*.toml"
+  - "rules/integrations/microsoft-365/**/*.toml"
```
Comment (rw-access) [inline .github/paths-labeller.yml:12]: should be `o365`.
can you check these?

File: .github/paths-labeller.yml [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+- "Platform: GCP":
+  - "rules/integrations/gcp/**/*.toml"
+- "Platform: Google Workspace":
+  - "rules/integrations/google-workspace/**/*.toml"
+- "Platform: Microsoft 365":
+  - "rules/integrations/microsoft-365/**/*.toml"
+- "Platform: Okta":
+  - "rules/integrations/okta/**/*.toml"
+- "python":
+  - "detection_rules/**/*.py"
```
Comment (rw-access) [inline .github/paths-labeller.yml:46]: what about KQL and other modules?

File: .github/paths-labeller.yml [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+- "RTA":
+  - "rta/**/*"
+- "Rule: Deprecation":
+  - "rules/_deprecated/**/*"
+# tests
+- "test-suite":
```
Comment (rw-access) [inline .github/paths-labeller.yml:52]: I think we can drop this one.
I was thinking of this as more like meta tests or a test exclusive PR.
Any new functionality that adds a corresponding test doesn't make sense to label with this in my opinion

File: .github/paths-labeller.yml [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+  - "etc/ecs_schemas/**/*"
+- "ML":
+  - "rules/ml/**/*"
+- "OS: Linux":
+  - "rules/linux/**/*.toml"
+- "OS: macOS":
+  - "rules/macos/**/*.toml"
+- "OS: Windows":
+  - "rules/windows/**/*.toml"
+- "Platform: AWS":
```
Comment (rw-access) [inline .github/paths-labeller.yml:29]: Think we should rename these labels from Platform to Integration?
Seems logical and like a good time to do it.

File: .github/paths-labeller.yml [hunk 10]
```diff
@@ -0,0 +1,53 @@
+---
+- "beats":
```
Comment (rw-access) [inline .github/paths-labeller.yml:2]: Should we just have a `schemas` label to cover beats + ecs?

File: .github/paths-labeller.yml (hunk 5 above)
Comment (brokensound77) [reply .github/paths-labeller.yml:6]: it should per glob rules

```python
glob.glob('detection-rules/**/*.md', recursive=True)
detection-rules-fork/CLI.md
detection-rules-fork/README.md
detection-rules-fork/CONTRIBUTING.md
detection-rules-fork/PHILOSOPHY.md
detection-rules-fork/rta/README.md
detection-rules-fork/docs/rule_insights.md
detection-rules-fork/docs/deprecating.md
detection-rules-fork/docs/experimental-machine-learning/DGA.md
detection-rules-fork/docs/experimental-machine-learning/problem-child.md
detection-rules-fork/docs/experimental-machine-learning/host-risk-score.md
detection-rules-fork/docs/experimental-machine-learning/url-spoof.md
detection-rules-fork/docs/experimental-machine-learning/readme.md
detection-rules-fork/docs/experimental-machine-learning/experimental-detections.md
detection-rules-fork/rules/README.md
```

File: .github/paths-labeller.yml (hunk 7 above)
Comment (brokensound77) [reply .github/paths-labeller.yml:46]: created `kibana-module` label and created entries:

```
- "kibana-module":
  - "kibana/**/*"
- "kql":
  - "kql/**/*"
```

File: .github/paths-labeller.yml (hunk 9 above)
Comment (brokensound77) [reply .github/paths-labeller.yml:29]: sounds good - changing now

File: .github/paths-labeller.yml (hunk 10 above)
Comment (brokensound77) [reply .github/paths-labeller.yml:2]: ok - I will remove those labels and update

File: .github/paths-labeller.yml (hunk 7 above)
Comment (rw-access) [reply .github/paths-labeller.yml]: oh i meant... i would rather not have module specific labels and would just call them all "python", regardless of where they are

Comment (brokensound77) [reply .github/paths-labeller.yml]: oh... so like it was? :nick-young-v2:

Comment (rw-access) [reply .github/paths-labeller.yml]: detection_rules/**
kibana/**
kql/**

etc

**Review by rw-access — APPROVED** [review]

**Pull request merged**